vision/pipeline.py

```diff
@@ -88,44 +93,92 @@ def run(self, prev_state):
+            except:
+                flags.setImg = False
+
+            try:
+                inFrame = self.module_location.isInFrame()
+            except:
+                inFrame = False
+                flags.isInFrame = False
 
             # only do more calculation if module is in the image
-            if self.module_location.is_in_frame():
-                center = self.module_location.get_center()  # center of module in image
-                depth = get_module_depth(
-                    depth_image, center
-                )  # depth of center of module
-
-                region = region_of_interest(
-                    depth_image, depth, center
-                )  # depth image sliced on underestimate bounds
-                orientation = get_module_orientation(
-                    region, center
-                )  # x and y tilt of module
-
-                bounds = get_module_bounds(
-                    color_image, center, depth
-                )  # overestimate of bounds
-                roll = get_module_roll(
-                    color_image[
-                        bounds[0][1] : bounds[3][1], bounds[0][0] : bounds[2][0], :
-                    ]
-                )  # roll of module
+            if inFrame:
+                try:
+                    center = self.module_location.getCenter()  # center of module in image
+                except:
+                    flags.getCenter = False
+
+                try:
+                    depth = get_module_depth(
+                        depth_image, center
+                    )  # depth of center of module
+                except:
+                    flags.get_module_depth = False
+
+                try:
+                    region = region_of_interest(
+                        depth_image, depth, center
+                    )  # depth image sliced on underestimate bounds
+                except:
+                    flags.region_of_interest = False
+                
+                try:
+                    orientation = get_module_orientation(
+                        region
+                    )  # x and y tilt of module
+                except:
+                    flags.get_module_orientation = False
+
+                try:
+                    bounds = getModuleBounds(
+                        color_image, center, depth
+                    )  # overestimate of bounds
+                except:
+                    flags.getModuleBounds = False
+
+                try:
+                    roll = get_module_roll(
+                        color_image[
+                            bounds[0][1] : bounds[3][1], bounds[0][0] : bounds[2][0], :
+                        ]
+                    )  # roll of module
+                except:
+                    flags.get_module_roll = False
 
                 # construct boundingbox for the module
                 box = BoundingBox(bounds, ObjectType.MODULE)
```

Comment:
I would add here that we wouldn't want to construct a module bounding box if the get_center function crashed, so you should add an if statement to check the flag.